### PR TITLE
HBASE-25599 Backport TestZKSecretWatcher improvement to branch-2.2

### PR DESCRIPTION
**Pull request by petersomogyi**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   4m 21s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ branch-2.2 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 15s |  branch-2.2 passed  |
| +1 :green_heart: |  compile  |   0m 54s |  branch-2.2 passed  |
| +1 :green_heart: |  checkstyle  |   1m 20s |  branch-2.2 passed  |
| +1 :green_heart: |  shadedjars  |   4m  0s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 39s |  branch-2.2 passed  |
| +0 :ok: |  spotbugs  |   3m 18s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m 15s |  branch-2.2 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 51s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 53s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 53s |  the patch passed  |
| -1 :x: |  checkstyle  |   1m 19s |  hbase-server: The patch generated 4 new + 1 unchanged - 0 fixed = 5 total (was 1)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m  2s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  25m 33s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 2.10.0 or 3.1.2 3.2.1.  |
| +1 :green_heart: |  javadoc  |   0m 38s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   3m 22s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 128m  7s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 35s |  The patch does not generate ASF License warnings.  |
|  |   | 191m 32s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2979/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2979 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 720ec8468e8a 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-home/workspace/Base-PreCommit-GitHub-PR_PR-2979/out/precommit/personality/provided.sh |
| git revision | branch-2.2 / ed9f8e6ce0 |
| Default Java | Oracle Corporation-1.8.0_282-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2979/1/artifact/out/diff-checkstyle-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2979/1/testReport/ |
| Max. process+thread count | 2838 (vs. ulimit of 12500) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2979/1/console |
| versions | git=2.20.1 maven=3.6.3 findbugs=3.1.11 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 10s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ branch-2.2 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 17s |  branch-2.2 passed  |
| +1 :green_heart: |  compile  |   0m 55s |  branch-2.2 passed  |
| +1 :green_heart: |  checkstyle  |   1m 21s |  branch-2.2 passed  |
| +1 :green_heart: |  shadedjars  |   4m  3s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 39s |  branch-2.2 passed  |
| +0 :ok: |  spotbugs  |   3m 25s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m 23s |  branch-2.2 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 52s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 55s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 55s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 19s |  hbase-server: The patch generated 0 new + 0 unchanged - 1 fixed = 0 total (was 1)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m  4s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  25m 45s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 2.10.0 or 3.1.2 3.2.1.  |
| +1 :green_heart: |  javadoc  |   0m 37s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   3m 28s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 131m 36s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 35s |  The patch does not generate ASF License warnings.  |
|  |   | 192m 32s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2979/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2979 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 20693e71d56d 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-home/workspace/Base-PreCommit-GitHub-PR_PR-2979/out/precommit/personality/provided.sh |
| git revision | branch-2.2 / a43fd04b6a |
| Default Java | Oracle Corporation-1.8.0_282-b08 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2979/2/testReport/ |
| Max. process+thread count | 3153 (vs. ulimit of 12500) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2979/2/console |
| versions | git=2.20.1 maven=3.6.3 findbugs=3.1.11 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/test/java/org/apache/hadoop/hbase/security/token/TestZKSecretWatcher.java [hunk 1]
```diff
@@ -140,35 +141,35 @@ public void testKeyUpdate() throws Exception {
...
-    LOG.debug("Master new current key: "+key2.getKeyId());
+    LOG.debug("Master new current key (key2) {}", key2);
     KEY_MASTER.rollCurrentKey();
     AuthenticationKey key3 = KEY_MASTER.getCurrentKey();
-    LOG.debug("Master new current key: "+key3.getKeyId());
+    LOG.debug("Master new current key (key3) {}", key3);
 
     // force expire the original key
-    key1.setExpiration(EnvironmentEdgeManager.currentTime() - 1000);
+    key1.setExpiration(EnvironmentEdgeManager.currentTime() - 100000);
```
Comment (Apache9) [inline hbase-server/src/test/java/org/apache/hadoop/hbase/security/token/TestZKSecretWatcher.java:166]: I know this is a backport, but mind explaining why change this value? I'm not familiar with the original improvement...

Comment (petersomogyi) [reply hbase-server/src/test/java/org/apache/hadoop/hbase/security/token/TestZKSecretWatcher.java:166]: I'm not sure of the original intention here as the -1000 should also make the key expired but it doesn't make any difference if the key expired long time ago.
If you prefer I can revert back this line.

https://github.com/apache/hbase/blob/branch-2.2/hbase-server/src/main/java/org/apache/hadoop/hbase/security/token/AuthenticationTokenSecretManager.java#L238-L256

Comment (Apache9) [reply hbase-server/src/test/java/org/apache/hadoop/hbase/security/token/TestZKSecretWatcher.java:166]: As you said this is a backport, what is the value on master branch?

Comment (petersomogyi) [reply hbase-server/src/test/java/org/apache/hadoop/hbase/security/token/TestZKSecretWatcher.java:166]: There is no difference on the master branch which would require the 100000 expiration.

**Review by Apache9 — APPROVED** [review]

**Pull request merged**